.github/workflows/build.yml

```diff
@@ -347,171 +335,113 @@ jobs:
+            test `ls -d /Applications/MNE-Python/*.app | wc -l` -eq 5 || exit 1
+            echo "Checking that the custom icon was set on the MNE folder in /Applications/MNE-Python"
+            test -f /Applications/MNE-Python/Icon$'\r' || exit 1
+          elif [[ "${{ runner.os }}" == "Linux" ]]; then
+            echo "Checking that menu shortcuts were created …"
+            pushd ~/.local/share/applications
+            ls -l || exit 1
+            echo "Checking for existence of .desktop files:"
+            ls MNE-Python*.desktop || exit 1
+            test `ls MNE-Python*.desktop | wc -l` -eq 5 || exit 1
+            echo ""
+
+            # … and patched to work around a bug in menuinst
+            echo "Checking that incorrect Terminal entries have been removed"
+            test `grep "Terminal=True"  MNE-Python*.desktop | wc -l` -eq 0 || exit 1
+            test `grep "Terminal=False" MNE-Python*.desktop | wc -l` -eq 0 || exit 1
+            echo ""
+
+            echo "Checking that Terminal entries are correct…"
+            test `grep "Terminal=true"  MNE-Python*.desktop | wc -l` -ge 1 || exit 1
+            test `grep "Terminal=false" MNE-Python*.desktop | wc -l` -ge 1 || exit 1
+            # Display their contents
+            for f in MNE-Python*.desktop; do echo "📂 $f:"; cat "$f"; echo; done
+            popd
+          fi
+
           echo "Checking the deployed environment variables were set correctly upon environment activation"
           mamba env config vars list
-          python -c "import os; assert 'PYTHONNOUSERSITE' in os.environ; assert os.environ['PYTHONNOUSERSITE'] == '1'"
-          python -c "import os; assert 'MAMBA_NO_BANNER' in os.environ; assert os.environ['MAMBA_NO_BANNER'] == '1'"
-
-          echo "Checking that menu shortcuts were created …"
-          pushd ~/.local/share/applications
-          ls -l
-          echo "Checking for existence of .desktop files:"
-          ls MNE-Python*.desktop
-          test `ls MNE-Python*.desktop | wc -l` -eq 5
-          echo ""
-
-          # … and patched to work around a bug in menuinst
-          echo "Checking that incorrect Terminal entries have been removed"
-          test `grep "Terminal=True"  MNE-Python*.desktop | wc -l` -eq 0
-          test `grep "Terminal=False" MNE-Python*.desktop | wc -l` -eq 0
-          echo ""
-
-          echo "Checking that Terminal entries are correct…"
-          test `grep "Terminal=true"  MNE-Python*.desktop | wc -l` -ge 1
-          test `grep "Terminal=false" MNE-Python*.desktop | wc -l` -ge 1
-
-          # Display their contents
-          for f in MNE-Python*.desktop; do echo "📂 $f:"; cat "$f"; echo; done
-          popd
+          if [[ "${{ runner.os }}" == "macOS" ]]; then
+            python -c "import os; x = os.getenv('CONDA_SUBDIR'); assert x == 'osx-64', f'CONDA_SUBDIR ({repr(x)}) != osx-64'" || exit 1
+          fi
+          # TODO: broken on Windows!
```

Comment:
These tests are present but not working on `main`:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AssertionError
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AssertionError
Running MNE's sys_info
```
because there is no `-e` there. This PR fixes the `-e`, but these have to be inside a conditional until we fix them on Windows (assuming they are actually meant to be fixed there)